File: WebClient SDK/src/default/workflow/krProcess/formatters/createCardStageTypeFormatter.ts
import { extension } from '@tessa/application';
import {
  IKrStageTypeFormatterContext,
  KrStageTypeFormatter,
  KrStageTypeFormattingHelper,
  StageTypeDescriptor,
  StageTypeDescriptors
} from '@tessa/platform';

@extension({ name: 'CreateCardStageTypeFormatter' })
export class CreateCardStageTypeFormatter extends KrStageTypeFormatter {
  //#region constants and static fields

  private static readonly _typeCaption = KrStageTypeFormattingHelper.formatPlainColumnName(
    'KrCreateCardStageSettingsVirtual',
    'TypeCaption'
  );
  private static readonly _templateCaption = KrStageTypeFormattingHelper.formatPlainColumnName(
    'KrCreateCardStageSettingsVirtual',
    'TemplateCaption'
  );
  private static readonly _modeName = KrStageTypeFormattingHelper.formatPlainColumnName(
    'KrCreateCardStageSettingsVirtual',
    'ModeName'
  );

  //#endregion

  //#region base overrides

  descriptors(): ReadonlyArray<StageTypeDescriptor> {
    return [StageTypeDescriptors.createCardDescriptor];
  }

  format(context: IKrStageTypeFormatterContext): void {
    const builder = { text: '' };

    this.appendString(
      builder,
      context.stageRow.tryGetString(CreateCardStageTypeFormatter._typeCaption),
      '',
      true,
      false,
      KrStageTypeFormatter.defaultSettingMax
    );

    this.appendString(
      builder,
      context.stageRow.tryGetString(CreateCardStageTypeFormatter._templateCaption),
      '',
      true,
      false,
      KrStageTypeFormatter.defaultSettingMax
    );

    this.appendString(
      builder,
      context.stageRow.tryGetString(CreateCardStageTypeFormatter._modeName),
      '',
      true,
      false,
      KrStageTypeFormatter.defaultSettingMax
    );

    context.displaySettings = builder.text;
  }

  //#endregion
}

File: Extensions/Tessa.Extensions.Default.Shared/Workflow/KrProcess/Formatters/CreateCardStageTypeFormatter.cs
#nullable enable

using System.Collections.Generic;
using System.Threading.Tasks;
using Tessa.Platform;
using Tessa.Platform.Storage;

namespace Tessa.Extensions.Default.Shared.Workflow.KrProcess.Formatters
{
    /// <summary>
    /// Форматтер типа этапа <see cref="StageTypeDescriptors.CreateCardDescriptor"/>.
    /// </summary>
    public sealed class CreateCardStageTypeFormatter :
        StageTypeFormatterBase
    {
        #region Base Overrides

        /// <inheritdoc />
        public override ValueTask FormatClientAsync(
            IStageTypeFormatterContext context)
        {
            FormatInternal(
                context,
                context.StageRow.Fields,
                true);

            return ValueTask.CompletedTask;
        }

        /// <inheritdoc />
        public override ValueTask FormatServerAsync(
            IStageTypeFormatterContext context)
        {
            FormatInternal(
                context,
                context.Settings!,
                false);

            return ValueTask.CompletedTask;
        }

        #endregion

        #region Private Methods

        private static void FormatInternal(
            IStageTypeFormatterContext context,
            IDictionary<string, object?> settings,
            bool isClient)
        {
            var sb = StringBuilderHelper.Acquire();

            AppendString(
                sb,
                settings.TryGet<string>(KrConstants.KrCreateCardStageSettingsVirtual.TypeCaption),
                null,
                true,
                limit: isClient ? DefaultSettingMax : -1);
            AppendString(
                sb,
                settings.TryGet<string>(KrConstants.KrCreateCardStageSettingsVirtual.TemplateCaption),
                null,
                true,
                limit: isClient ? DefaultSettingMax : -1);
            AppendString(
                sb,
                settings.TryGet<string>(KrConstants.KrCreateCardStageSettingsVirtual.ModeName),
                null,
                true,
                limit: isClient ? DefaultSettingMax : -1);

            context.DisplaySettings = sb.ToStringAndRelease();
        }

        #endregion
    }
}
